nautobot/core/templates/inc/nav_menu.html

```diff
@@ -2,27 +2,57 @@
 {% load registry %}
 {% load helpers %}
 {% registry %}
-<nav class="navbar navbar-default navbar-fixed-top">
+<nav class="navbar navbar-default navbar-fixed-left navbar-inverse">
     <div class="container-fluid">
         <div class="navbar-header">
-            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
-                <span class="sr-only">Toggle navigation</span>
-                <span class="icon-bar"></span>
-                <span class="icon-bar"></span>
-                <span class="icon-bar"></span>
-            </button>
             <a class="navbar-brand" href="{% url 'home' %}">
                 <img src="{% custom_branding_or_static 'logo' 'img/nautobot_logo.svg' %}" height="30" />
             </a>
         </div>
 
-        <div id="navbar" class="navbar-collapse collapse">
+        <div id="navbar">
+            {% if request.user.is_authenticated or not "HIDE_RESTRICTED_UI"|settings_or_config %}
+            <form action="{% url 'search' %}" method="get" class="navbar-form" id="navbar_search" role="search">
+                <div class="input-group">
+                    <input type="text" name="q" class="form-control" placeholder="Search {{ settings.BRANDING_TITLE }}">
+                    <span class="input-group-btn">
+                        <button type="submit" class="btn btn-primary">
+                            <i class="mdi mdi-magnify"></i>
+                        </button>
+                    </span>
+                </div>
+            </form>
+            {% endif %}
+
+            <ul class="nav navbar-nav">
+                {% if request.user.is_authenticated %}
+                <li class="dropdown">
+                    <a href="#" class="dropdown-toggle" data-toggle="dropdown" title="{{ request.user }}" role="button" aria-haspopup="true" aria-expanded="false">
+                        <i class="mdi mdi-account"></i>
+                        <span id="navbar_user">{{ request.user|truncatechars:"30" }}</span>
```

Comment:
I'm only seeing 29 characters show up:

![image](https://github.com/nautobot/nautobot/assets/75227981/7d9ffb8a-5039-4a06-b9b3-ce9cfb7c1f12)

And fewer if there's any capital letters since the font isn't fixed width. Maybe we need to drop this down a bit? 22 works best for my browser:

![image](https://github.com/nautobot/nautobot/assets/75227981/fedce610-fea5-4fa4-8bc0-c58d4a242ae5)

Reply:
I see, I overlooked this part of the template before. Anyway, if this text needs to be truncated to always occupy one line, it can be done with CSS `text-overflow: ellipsis;`, that way we don't have to give an explicit maximum characters count. I did it in 856735c5a3e2ce3c17e8678c7870e25465ef8e3a

Reply:
Looks like this:

| Long user name | Long user name mouse hover |
| - | - |
| ![image](https://github.com/nautobot/nautobot/assets/117445994/877d1b0d-ec42-423f-83f8-a673c4c9d90f) | ![image](https://github.com/nautobot/nautobot/assets/117445994/11216b0f-9141-4ba6-90b1-7b6191095dc2) |